krabs/krabs/ut.hpp

```diff
@@ -194,11 +194,33 @@ namespace krabs { namespace details {
         const EVENT_RECORD &record,
         const krabs::trace<krabs::details::ut> &trace)
     {
-        for (auto &provider : trace.providers_) {
+        auto found = false;
+
+        // skip event trace header events
+        if (record.EventHeader.ProviderId == EventTraceGuid) {
```

Comment:
under what conditions will this be true? should we only do this if we weren't able to find a manifest provider below?

Reply:
EventTraceGuid is the 'log header event'.
https://docs.microsoft.com/en-us/windows/win32/etw/eventtraceevent

All of the MSDN examples skip this event. I followed suit - but really mostly so that I could skip these events while debugging and hit the first 'real' event sooner.

I also had a vague idea that this would make it more performant. i.e. skipping unnecessary comparisons in the loops.
But on reflection, as these are header-only events, it makes more sense overall to not include this check.

Reply:
Plus technically someone could subscribe to the header events.